packages/next-app/src/lib/gql.ts

```diff
@@ -1,16 +1,42 @@
 import { GraphQLClient } from "graphql-request";
-import {ENDPOINTS} from "@balancer-pool-metadata/balancer-gql/codegen";
-// import { getSdkWithHooks as mainnetSdk  } from "@balancer-pool-metadata/balancer-gql/src/gql/__generated__/mainnet";
-import { getSdkWithHooks as goerliSdk  } from "@balancer-pool-metadata/balancer-gql/src/gql/__generated__/goerli";
-// import { getSdkWithHooks as polygonSdk  } from "@balancer-pool-metadata/balancer-gql/src/gql/__generated__/polygon";
-// import { getSdkWithHooks as arbitrumSdk  } from "@balancer-pool-metadata/balancer-gql/src/gql/__generated__/arbitrum";
+import {
+  ENDPOINTS,
+  Network,
+} from "@balancer-pool-metadata/balancer-gql/codegen";
+import { getSdkWithHooks as mainnetSdk } from "@balancer-pool-metadata/balancer-gql/src/gql/__generated__/mainnet";
+import { getSdkWithHooks as goerliSdk } from "@balancer-pool-metadata/balancer-gql/src/gql/__generated__/goerli";
+import { getSdkWithHooks as polygonSdk } from "@balancer-pool-metadata/balancer-gql/src/gql/__generated__/polygon";
+import { getSdkWithHooks as arbitrumSdk } from "@balancer-pool-metadata/balancer-gql/src/gql/__generated__/arbitrum";
 
+const currentNetwork =
+  typeof localStorage !== "undefined"
+    ? localStorage.getItem("networkId") ?? "1"
+    : "1";
```

Comment:
I imagine this value will update if you change the network selector, but it probably won't be read again because this value would've already been evaluated the first time the browser loads this part of the code. Have you tested this?

Reply:
When the index page is loaded, we save the `networkId` in the local storage. 

Whenever the `networkId` changes (with the user shifting the network), we refresh the page with `Router.reload()`.

After refreshing the page we set the current networkId on the local storage. As the gqp interface is called after setting the networkId on the local storage, we always have a value for the networkId.